### ISSUE TITLE
Duplicate tokens show up on rotate

### ISSUE THREAD

**Issue by massivespace** [splitwise/TokenAutoComplete#157]
I have a problem I'm having trouble finding a work around for.

I'm running Android 4.4 with latest TokenAutoComplete.  I have a fragment with a token autocomplete view in it that I add tokens to in the onResume method.  Initially, I just cleared the tokens by calling TokenCompleteTextView.clear(), then tried TokenCompleteTextView.getObjects.clear().  In both cases, when I rotate the device, onResume got called, and I ended up adding double tokens every time, additively.  

I then thought I'd put a check in before adding a token to make sure the getObjects() call returned an array length zero before adding tokens.  To my surprise, it was returning an array length of 0 while called from the onResume() method in the fragment, so I added 3 tokens, and then it showed 6.  

I tried placing the code in several other methods with no success.  The only way I can prevent duplicate tokens showing is to post a runnable with a 100 ms delay (smaller values were not successful, and of course this will vary per device).  In the runnable, I clear and repopulate the objects by calling addObject.

So in summary, create a fragment with a Token view in it, then clear the objects and call addObject() in the onResume method.  Rotate device, and bug presents itself.

**Comment by mgod** [comment]
I'm on vacation at the moment, so can't pull up the code to double check, but I would look at adding your tokens in onCreate instead of onResume or look at overriding getSerializableObjects to return an empty array. 



—

**Comment by massivespace** [comment]
Hi Marshall, 
Thanks for replying.  Since this is a fragment,  onCreate gets called before onCreateView, and I also tried doing this in onCreateView and onViewStateRestored.  I will try overriding the serializable methods though,  thanks! 
Chris
On Aug 3, 2015 11:16 AM, Marshall Weir <notifications@github.com> wrote:I'm on vacation at the moment, so can't pull up the code to double check, but I would look at adding your tokens in onCreate instead of onResume or look at overriding getSerializableObjects to return an empty array. 



—

**Comment by luaz** [comment]
Whenever I rotate the screen, the field become empty again.

**Comment by massivespace** [comment]
Thanks luaz! I actually ended up returning an empty set in the GetSerializableObjects override, but later I override the equal method like you so that allowDuplicates=false would work,  so I'll try to remove the GetSerializableObjects hack now. Good info!
